pyface/ui/wx/util/event_loop_helper.py

```diff
@@ -0,0 +1,151 @@
+
+        """
+        yield
+        for i in range(repeat):
+            self.gui_app.ProcessPendingEvents()
+
+    def event_loop_until_condition(self, condition, start=None, timeout=10.0):
+        """ Runs the real event loop until the provided condition evaluates
+        to True.
+
+        Raises ConditionTimeoutError if the timeout occurs before the condition
+        is satisfied.
+
+        Parameters
+        ----------
+        condition : callable
+            A callable to determine if the stop criteria have been met. This
+            should accept no arguments.
+
+        start : callable
+            A callable to use in order to start the event loop. Default is
+            to create a small frame or reuse the top level window if it is
+            still available.
+
+        timeout : float
+            Number of seconds to run the event loop in the case that the trait
+            change does not occur.
+
+        """
+        def handler(event):
+            if condition():
+                self.gui_app.Exit()
+
+        # Make sure we don't get a premature exit from the event loop.
+        with dont_quit_when_last_window_closed(self.gui_app):
+            condition_timer = wx.Timer(self._handler)
+            timeout_timer = wx.Timer(self._handler)
+            self._handler.Bind(wx.EVT_TIMER, handler, condition_timer)
+            self._handler.Bind(
+                wx.EVT_TIMER, lambda event:self.gui_app.Exit(), timeout_timer)
+            timeout_timer.Start(int(timeout * 1000), True)
+            condition_timer.Start(50)
+            try:
+                if start is None:
+                    self._start_event_loop()
+                else:
+                    start()
+                if not condition():
+                    raise ConditionTimeoutError(
+                        'Timed out waiting for condition')
+            finally:
+                timeout_timer.Stop()
+                condition_timer.Stop()
+
+    @contextlib.contextmanager
+    def delete_window(self, window, timeout=2.0):
+        """ Runs the real event loop until the window provided has been deleted.
+
+        Parameters
+        ----------
+        window :
+            The toolkit window widget whose deletion will stop the event loop.
+
+        timeout : float
+            Number of seconds to run the event loop in the case that the
+            widget is not deleted.
+
+        Raises
+        ------
+        ConditionTimeoutError:
+            Raised on timeout.
+
+        """
+        timer = wx.Timer(self._handler)
+        self._handler.Bind(
+            wx.EVT_TIMER, lambda event:self.gui_app.Exit(), timer)
+        yield
+        timer.Start(int(timeout * 1000), True)
+        self._start_event_loop()
+        if not timer.IsRunning():
+            # We exited the event loop on timeout
+            raise ConditionTimeoutError(
+                'Could not destroy widget before timeout: {!r}'.format(widget))

```

Comment:
`widget` is not defined here.  `window` instead?

Reply:
That is probably true